src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+									}
+									else if (eventSymbol != null)
+									{
+										// If a binding is inside a DataTemplate, the binding root in the case of an x:Bind is
+										// the DataContext, not the control's instance.
+										var isInsideDataTemplate = IsMemberInsideFrameworkTemplate(member.Owner);
+
+										void writeEvent(string ownerPrefix)
+										{
+											if (eventSymbol.Type is INamedTypeSymbol delegateSymbol)
+											{
+												var parms = delegateSymbol
+													.DelegateInvokeMethod
+													.Parameters
+													.Select(p => member.Value + "_" + p.Name)
+													.JoinBy(",");
+
+												var eventSource = ownerPrefix.HasValue() ? ownerPrefix : "this";
+
+												//
+												// Generate a weak delegate, so the owner is not being held onto by the delegate. We can
+												// use the WeakReferenceProvider to get a self reference to avoid adding the cost of the
+												// creation of a WeakReference.
+												//
+												writer.AppendLineInvariant($"var {member.Member.Name}_{member.Value}_That = ({eventSource} as global::Uno.UI.DataBinding.IWeakReferenceProvider).WeakReference;");
+												writer.AppendLineInvariant($"{closureName}.{member.Member.Name} += ({parms}) => ({member.Member.Name}_{member.Value}_That.Target as {_className.className})?.{member.Value}({parms});");
+											}
+											else
+											{
+												GenerateError(writer, $"{eventSymbol.Type} is not a supported event");
+											}
+										}
+
+										if (!isInsideDataTemplate)
+										{
+											writeEvent("");
+										}
+										else if (_className.className != null)
+										{
+											_hasLiteralEventsRegistration = true;
+											writer.AppendLineInvariant($"{closureName}.RegisterPropertyChangedCallback(");
+											using (writer.BlockInvariant($"global::Uno.UI.Xaml.XamlInfo.XamlInfoProperty, (s, p) =>"))
+											{
+												using (writer.BlockInvariant($"if (global::Uno.UI.Xaml.XamlInfo.GetXamlInfo({closureName})?.Owner is {_className.className} owner)"))
+												{
+													writeEvent("owner");
+												}
+											}
+											writer.AppendLineInvariant($");");
+										}
+										else
+										{
+											GenerateError(writer, $"Unable to use event {member.Member.Name} without a backing class (use x:Class)");
+										}
+									}
+									else
+									{
+										GenerateError(
+											writer,
+											$"Property {member.Member.PreferredXamlNamespace}:{member.Member} is not available on {member.Member.DeclaringType?.Name}, value is {member.Value}"
+										);
+									}
+								}
+							}
+						}
+					}
+
+					if (_isUiAutomationMappingEnabled)
+					{
+						// Prefer using the Uid or the Name if their value has been explicitly assigned
+						var assignedUid = uidMember?.Value?.ToString();
+						var assignedName = nameMember?.Value?.ToString();
+
+						if (assignedUid.HasValue())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['assignedUid' is null on at least one execution path.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)